libs/framework/include/celix/dm/Component_Impl.h

```diff
@@ -39,21 +39,25 @@ Component<T>::~Component() {
 
 template<class T>
 template<class I>
-Component<T>& Component<T>::addInterfaceWithName(const std::string serviceName, const std::string version, const Properties properties) {
+Component<T>& Component<T>::addInterfaceWithName(const std::string &serviceName, const std::string &version, const Properties &properties) {
     if (!serviceName.empty()) {
         //setup c properties
         celix_properties_t *cProperties = properties_create();
         properties_set(cProperties, CELIX_FRAMEWORK_SERVICE_LANGUAGE, CELIX_FRAMEWORK_SERVICE_CXX_LANGUAGE);
         for (const auto& pair : properties) {
-            properties_set(cProperties, (char *) pair.first.c_str(), (char *) pair.second.c_str());
+            char *key = nullptr;
+            char *value = nullptr;
+            copyString(pair.first, &key);
+            copyString(pair.second, &value);
+            properties_set(cProperties, key, value);
```

Comment:
note that properties_set does een copy of the provided key/value, so now you have a double copy

Reply:
Nice find.